mlflow/models/evaluation/base.py

```diff
@@ -1334,12 +1337,22 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
         evaluator_name_to_conf_map,
     ) = _normalize_evaluators_and_evaluator_config_args(evaluators, evaluator_config)
 
-    dataset = EvaluationDataset(
-        data,
-        targets=targets,
-        path=dataset_path,
-        feature_names=feature_names,
-    )
+    from mlflow.data.pyfunc_dataset_mixin import PyFuncConvertibleDatasetMixin
+
+    if isinstance(data, Dataset) and issubclass(data.__class__, PyFuncConvertibleDatasetMixin):
+        dataset = data.to_evaluation_dataset(dataset_path, feature_names)
+        if evaluator_name_to_conf_map and "default" in evaluator_name_to_conf_map:
+            context = evaluator_name_to_conf_map["default"].get("metric_prefix", None)
+        else:
+            context = None
+        mlflow.log_input(data, context)
```

Comment:
We're doing this outside of line 1357 (`_start_run_or_reuse_active_run()`), so the run management might not be correct here. Can we move this input logging down into the context created on 1357?

Reply:
Done. Also changed to client API